### ISSUE TITLE
windows support

### ISSUE THREAD

**Issue by BoPeng** [vatlab/SOS#172]
Tested sos under windows just because I have a windows machine at hand. It turns out that sos supports windows except for the `blassings` module, which requires the `ncurse` terminal support that is unavailable under windows.

The other problem is that the `sos` script without extension is difficult to use (require `python /path/to/sos`) so some sort of `py2exe` or `cx_Freeze` is required. Then it turns out that these two tools do not support python 3.5 well.

We perhaps also need to add action to execute windows `bat` file. We are using both `run` and `bash` for bash scripts. I suppose we should use `bash` to long bash scripts, `run` for simple list of commands that are compatible with both windows bat and linux shell. 

The bottom line is that we can make sos windows compatible without much effort and perhaps some users can find creative use of this tool under windows.

**Comment by BoPeng** [comment]
Even if we cannot get the sos command line (batch mode) work perfectly, the jupyter notebook would be valuable for working with R and Python under windows.

**Comment by BoPeng** [comment]
sos and sos-runner executables are fixed... which is nice.

Bessings needs to replaced by colorama, not sure how well it would work. 

Most tests fail because of windows linux path problem.

Jupyter notebook seems to be working...

**Comment by BoPeng** [comment]
Windows path problems are very difficult to fix, because for example all `relpath` calls would fail if the user directory and sos directory are on different drives. I got errors such as "path is on mount "X:", start on mount "C:"

**Comment by BoPeng** [comment]
There is a even larger problem here. Under linux `env.sos_dict` is shared to the child process so that the workflow steps is executed with the parental namespace. Under linux, a fresh subprocess is created so `env.sos_dict` is empty. I would have to explicitly transfer `env.sos_dict` to the child process.

**Comment by BoPeng** [comment]
The resource sharing problem has been resolved but there are individual path problems. I will close this ticket and open tickets for individual problems.
